src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1735,6 +1736,17 @@ source = do
 linkTitle :: PandocMonad m => MarkdownParser m Text
 linkTitle = quotedTitle '"' <|> quotedTitle '\''
 
+-- Github wiki style link, with optional title
+githubWikiLink :: PandocMonad m => MarkdownParser m (F Inlines)
+githubWikiLink = try $ do
+  guardEnabled Ext_markdown_github_wikilink
+  string "[["
+  skipSpaces
+  let allUntil = manyTillChar (satisfy $ const True)
+  part1 <- allUntil (oneOf "|]")
+  part2 <- (char ']' >> return part1) <|> allUntil (string "]]")
```

Comment:
`allUntil (string "]]")`: note that you generally want something like `(try (string "]]"))` in this context, because otherwise your parser will fail when it encounters a single `]` with something else after it.  (Maybe that's what you want in this case, though?)